### ISSUE TITLE
Add Microsoft.AspNetCore.HttpOverrides to CreateDefaultBuilder()

### ISSUE THREAD

**Issue by jchannon** [aspnet/MetaPackages#253]
`CreateDefaultBuilder` calls `UseIISIntegraiton` but obviously if this isn't used on a OSX/Linux `UseHttpMethodOverride` isn't invoked to handle headers like `X-Forwarded-Proto`. 

`UseHttpMethodOverride` should not be a default feature just for IIS users only.

Reference : https://github.com/aspnet/BasicMiddleware/tree/dev/src/Microsoft.AspNetCore.HttpOverrides

**Comment by Tratcher** [comment]
You're referring to UseForwardedHeaders, not UseHttpMethodOverride, correct?

This is intentional. UseForwardedHeaders is actually quite dangerous as it allows local values like scheme and remote IP to be overridden by remote data. This must not be used unless you have some level of trust in said data, e.g. you know you have a reverse proxy that sets these specific headers. This is why it's added by UseIISIntegraiton only when running behind IIS, it knows what headers will be provided by AspNetCoreModule.

There's an acknowledged gap where we have no UseNginxIntegration to mirror UseIISIntegraiton. We'd want this to provide much of the same functionality such as lighting up only when behind nginx, adding UseForwardedHeaders, handling path base, etc.. This would be paired with docs on the recommended nginx config file to use to match the server settings.

**Comment by jchannon** [comment]
Yeah exactly right. We have nginx and assumed the UseForwardHeaders was a
default behaviour in the runtime somehow.

Out of interest even with IIS how do you know that iis send the forwarded
header and not some cheeky hacker.

**Comment by Tratcher** [comment]
We use environment variables at startup to detect that we're running behind IIS/ANCM, and in ANCM we always have it set those.
